lib/trollop.rb

```diff
@@ -635,170 +575,322 @@ def cloaker(&b)
+                              when Array
+                                default.join(", ")
+                              else
+                                default.to_s
+                              end
+                  
+                  if desc =~ /\.$/
+                    " (Default: #{default_s})"
+                  else
+                    " (default: #{default_s})"
+                  end
+                end
+    return desc + defstring
+    
+  end
+
+  ## Provide a way to register symbol aliases to the Parser
+  def self.register_alias(*alias_keys)
+    alias_keys.each do |alias_key|
+      # pass in the alias-key and the class
+      Parser.register(alias_key, self)
     end
+  end
+  
+  ## Factory class methods ...
+  
+  # Determines which type of object to create based on arguments passed
+  # to +Trollop::opt+.  This is trickier in Trollop, than other cmdline
+  # parsers (e.g. Slop) because we allow the +default:+ to be able to
+  # set the option's type.
+  def self.create(name, desc="", opts={}, settings={})
 
-    type_from_default =
-      case disambiguated_default
-      when Integer     then :int
-      when Numeric     then :float
-      when TrueClass,
-           FalseClass  then :flag
-      when String      then :string
-      when IO          then :io
-      when Date        then :date
-      when Array
-        if opts[:default].empty?
-          if opts[:type]
-            raise ArgumentError, "multiple argument type must be plural" unless MULTI_ARG_TYPES.include?(opts[:type])
-            nil
-          else
-            raise ArgumentError, "multiple argument type cannot be deduced from an empty array for '#{opts[:default][0].class.name}'"
-          end
-        else
-          case opts[:default][0]    # the first element determines the types
-          when Integer then :ints
-          when Numeric then :floats
-          when String  then :strings
-          when IO      then :ios
-          when Date    then :dates
-          else
-            raise ArgumentError, "unsupported multiple argument type '#{opts[:default][0].class.name}'"
-          end
-        end
-      when nil         then nil
-      else
-        raise ArgumentError, "unsupported argument type '#{opts[:default].class.name}'"
-      end
+    opttype = Trollop::Parser.registry_getopttype(opts[:type])
+    opttype_from_default = get_klass_from_default(opts, opttype)
 
-    raise ArgumentError, ":type specification and default type don't match (default type is #{type_from_default})" if opts[:type] && type_from_default && opts[:type] != type_from_default
+    raise ArgumentError, ":type specification and default type don't match (default type is #{opttype_from_default.class})" if opttype && opttype_from_default && (opttype.class != opttype_from_default.class)
 
-    opts[:type] = opts[:type] || type_from_default || :flag
+    @optinst = (opttype || opttype_from_default || Trollop::BooleanOption.new)
```

Comment:
is there a reason `@optinst` is a class level variable?

Reply:
I don't think there was a reason, will clean that up.